src/vmod_xkey.c

```diff
@@ -602,14 +608,21 @@ vmod_event(VRT_CTX, struct vmod_priv *priv, enum vcl_event_e e)
 	switch (e) {
 	case VCL_EVENT_LOAD:
 		AZ(pthread_mutex_lock(&mtx));
-		if (n_init == 0)
+		if (n_init == 0) {
 #ifdef HAVE_ENUM_EXP_EVENT_E
 			xkey_cb_handle =
 			    EXP_Register_Callback(xkey_cb, NULL);
 #else
 			xkey_cb_handle = ObjSubscribeEvents(xkey_cb, NULL,
 			    OEV_INSERT|OEV_EXPIRE);
 #endif
+			if (vsc == NULL) {
+				AZ(vsc_seg);
+				vsc = VSC_xkey_New(NULL, &vsc_seg, "");
+			}
+			AN(vsc);
```

Comment:
This was copied from vmod_debug. Maybe the code here should be more robust and handle the case where VSC_xkey_New returns NULL (if that's possible).

Reply:
but that can't happen with `n_init == 0`, can it?

Reply:
After learning more about vmod_xkey and Varnish's VMOD interface over the last few days, I've come to that conclusion as well.